packages/blade/src/components/Checkbox/_decisions/decisions.md

```diff
@@ -0,0 +1,373 @@
+
+We can't pass `isRequired` & `isInvalid`
+
+Because we don't know that:
+
+- Does user require that at least one checkbox is checked?
+- Does user require that all checkboxes are to be checked?
+
+When we say a checkbox group is required we generally mean that any of the checkboxes in a group has to be checked but if we pass down `isRequired` to all the checkboxes it would make the behaiour as all the checkboxes needs to be ticked. On the other hand there could be cases where we want all the checkboxes to be required not just one.
+
+Similarly with `isInvalid` each checkbox should be independent of it's validation state. We don't know if the consumer wants `some` checkboxes or `all` checkboxes to be required.
+
+Ultimately IMO the validation logic should be handled by the consumer side depending on their requirements, and they should provide proper `helperText` to communicate the intent.
+
+Although we will still have the `isRequired` `isInvalid` prop to show visual changes like showing the `necessity indication` or `negative error text`
+
+**Example of user managed validation**
+
+Example of only one checkbox needs to be checked:
+
+```tsx
+function AnyOneHasToBeSelected() {
+  const [selected, setSelected] = React.useState([]);
+
+  return (
+    <CheckboxGroup
+      label="Do you love any of the fruites?"
+      helperText="You can select multiple"
+      value={selected}
+      onChange={setSelected}
+      isRequired
+      isInvalid={selected.length === 0 ? true : false}
+    >
+      <Checkbox value="apple">apple</Checkbox>
+      <Checkbox value="mango">mango</Checkbox>
+      <Checkbox value="orange">orange</Checkbox>
+      <Checkbox value="banana">banana</Checkbox>
+    </CheckboxGroup>
+  );
+}
+```
+
+Example of every checkboxes needs to be checked:
+
+```tsx
+function AllOfThemHasToBeSelected() {
+  const [selected, setSelected] = React.useState([]);
+
+  return (
+    <CheckboxGroup
+      label="Do you agree to all the terms?"
+      helperText="Select all"
+      value={selected}
+      onChange={setSelected}
+      isRequired
+      isInvalid={selected.length === 3 ? true : false}
+    >
+      <Checkbox value="use-ts">I will use TS</Checkbox>
+      <Checkbox value="no-any">I won't use any</Checkbox>
+      <Checkbox value="use-js">I won't use JS</Checkbox>
+    </CheckboxGroup>
+  );
+}
+```
+
+And lastly users can also individually set `isRequired`, `isInvalid` props on the <Checkbbox \/> components instead of the parent CheckboxGroup for greater fine control.
+
+```tsx
+function IndividualCheckboxValidation() {
+  return (
+    <CheckboxGroup label="Do you agree to all the terms?" helperText="Select all">
+      <Checkbox value="use-ts" isRequired>
+        I will use TS
+      </Checkbox>
+      <Checkbox value="no-any">I won't use any (you can use it)</Checkbox>
+      <Checkbox value="use-js" isRequired>
+        I won't use JS
+      </Checkbox>
+    </CheckboxGroup>
+  );
+}
+```
+
+## Component Architecture
```

Comment:
Good details but I'd suggest removing these from the decision doc since this might attract a lot of feedbacks & suggestions that might not be relevant at API decision phase. We can review this in the implementation PR instead and you might realise some other things while implementing that you might not be able to at this time which could change this architecture too.